### PR TITLE
[4.0] Move accordion/collapse to JLayouts

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue # .

### Summary of Changes

This PR moves the accordion HTML to layouts, allowing people to finally override them.

### Testing Instructions

- Apply PR
- Create a new menu item and open the modal "Menu Item Type"

### Expected result

The accordion should work the same as before the PR

### CODE REVIEW

**Comment by brianteeman** [comment]
layouts that just have closing tags seems massively wrong to me - there must be a better way to do that

File: layouts/joomla/accordion/startaccordionscript.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+$options  = empty($displayData['options']) ? '' : $displayData['options'];
+$selector = empty($displayData['selector']) ? '' : $displayData['selector'];
+$onShow   = empty($displayData['onShow']) ? '' : $displayData['onShow'];
+$onShown  = empty($displayData['onShown']) ? '' : $displayData['onShown'];
+$onHide   = empty($displayData['onHide']) ? '' : $displayData['onHide'];
+$onHidden = empty($displayData['onHidden']) ? '' : $displayData['onHidden'];
+
+$script = [];
+$script[] = "jQuery(function($){";
```
Comment (dgt41) [inline layouts/joomla/accordion/startaccordionscript.php]: NOPE! The script needs to be in a file not inline. You can get an idea how such thing can be accomplished by looking at https://github.com/joomla/joomla-cms/pull/11922

Comment (C-Lodder) [inline layouts/joomla/accordion/startaccordionscript.php]: Just using the same approach as the tabs

**Comment by C-Lodder** [comment]
@brianteeman if I knew a way...

File: layouts/joomla/accordion/startaccordionscript.php (hunk 1 above)
Comment (dgt41) [inline layouts/joomla/accordion/startaccordionscript.php]: We have to get rid of all the inline script to make the core CSP strict compatible

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 1018ad99ace1675cd6f21d2f357f3b8df5c8c784<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14768">issues.joomla.org/tracker/joomla-cms/14768</a>.</sub>

**Comment by dgt41** [comment]
@C-Lodder layouts support sub-layouts https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/layout/file.php#L650
But you'll have to make a B/C break (or not, haven't look the code here TBH)

**Comment by mbabker** [comment]
👎 

All JHtml methods can be overridden through the use of its API for defining what method a key should call as demonstrated in https://github.com/mbabker/bs3-demo/blob/master/plg_system_bootstrap3/bootstrap3.php

We should not be having JHtml methods (overridable by API design) be heavily using JLayout.  This potentially creates a confusing chain of figuring out what has been overridden (and therefore what's actually in use on the site).

If the intent is to shuffle markup out of JHtml methods, the helpers should be deprecated and JLayout should just be used directly.

**Comment by C-Lodder** [comment]
@mbabker why has it been done for tabs then?

**Comment by mbabker** [comment]
I can't say I liked it then either.  But not the first time I've been ignored.

**Pull request closed without merging**